SocketChannelTest.testAcceptFailsWith{EMFILE,ENFILE} flaky

just got this in Docker (Swift 4.1):

```
Test Case 'SocketChannelTest.testAcceptFailsWithEMFILE' started at 2018-05-02 14:15:46.819
swift-nio/Tests/NIOTests/SocketChannelTest.swift:155: error: SocketChannelTest.testAcceptFailsWithEMFILE : XCTAssertNoThrow failed: threw error "ioOnClosedChannel" - 
Assertion failed: file swift-nio/Sources/NIO/BaseSocketChannel.swift, line 865
Current stack trace:
0    libswiftCore.so                    0x00007fc573879750 _swift_stdlib_reportFatalErrorInFile + 221
1    libswiftCore.so                    0x00007fc5735900f0 <unavailable> + 1442032
2    libswiftCore.so                    0x00007fc5738206a2 <unavailable> + 4130466
3    libswiftCore.so                    0x00007fc573821869 <unavailable> + 4135017
4    libswiftCore.so                    0x00007fc57358f496 <unavailable> + 1438870
5    libswiftCore.so                    0x00007fc5738202fb <unavailable> + 4129531
6    libswiftCore.so                    0x00007fc57358f496 <unavailable> + 1438870
7    libswiftCore.so                    0x00007fc573727679 <unavailable> + 3110521
8    libswiftCore.so                    0x00007fc57358ece0 _assertionFailure(_:_:file:line:flags:) + 44
9    swift-nioPackageTests.xctest       0x000055d870318a84 <unavailable> + 932484
10   swift-nioPackageTests.xctest       0x000055d87031a29e <unavailable> + 938654
11   swift-nioPackageTests.xctest       0x000055d8708897ba <unavailable> + 6637498
12   swift-nioPackageTests.xctest       0x000055d87089ccfc <unavailable> + 6716668
13   swift-nioPackageTests.xctest       0x000055d870711e8a <unavailable> + 5099146
14   swift-nioPackageTests.xctest       0x000055d87089cd5b <unavailable> + 6716763
15   swift-nioPackageTests.xctest       0x000055d87038671a <unavailable> + 1382170
16   swift-nioPackageTests.xctest       0x000055d870386858 <unavailable> + 1382488
17   swift-nioPackageTests.xctest       0x000055d8703319ec <unavailable> + 1034732
18   swift-nioPackageTests.xctest       0x000055d870394f91 <unavailable> + 1441681
19   swift-nioPackageTests.xctest       0x000055d87038ca77 <unavailable> + 1407607
20   swift-nioPackageTests.xctest       0x000055d8703916d1 <unavailable> + 1427153
21   swift-nioPackageTests.xctest       0x000055d87038ca97 <unavailable> + 1407639
22   swift-nioPackageTests.xctest       0x000055d870391601 <unavailable> + 1426945
23   swift-nioPackageTests.xctest       0x000055d87030f3e6 <unavailable> + 893926
24   swift-nioPackageTests.xctest       0x000055d87039166b <unavailable> + 1427051
25   swift-nioPackageTests.xctest       0x000055d870386dde <unavailable> + 1383902
26   swift-nioPackageTests.xctest       0x000055d87038bad7 <unavailable> + 1403607
27   swift-nioPackageTests.xctest       0x000055d87038e9e4 <unavailable> + 1415652
28   swift-nioPackageTests.xctest       0x000055d870394aaa <unavailable> + 1440426
29   swift-nioPackageTests.xctest       0x000055d87038efef <unavailable> + 1417199
30   swift-nioPackageTests.xctest       0x000055d87043aeb1 <unavailable> + 2121393
31   swift-nioPackageTests.xctest       0x000055d87043b7e3 <unavailable> + 2123747
32   swift-nioPackageTests.xctest       0x000055d87043b889 <unavailable> + 2123913
33   libpthread.so.0                    0x00007fc573d97182 <unavailable> + 33154
34   libc.so.6                          0x00007fc571cb8410 clone + 109
Exited with signal code 4
```

and again

```
swift-nio/Tests/NIOTests/SocketChannelTest.swift:155: error: SocketChannelTest.testAcceptFailsWithENFILE : XCTAssertNoThrow failed: threw error "ioOnClosedChannel" - 
Assertion failed: file swift-nio/Sources/NIO/BaseSocketChannel.swift, line 865
```

--- a/Tests/NIOTests/SocketChannelTest.swift
+++ b/Tests/NIOTests/SocketChannelTest.swift
@@ -148,11 +148,13 @@ public class SocketChannelTest : XCTestCase {
         let serverChannel = try ServerSocketChannel(serverSocket: socket, eventLoop: group.next() as! SelectableEventLoop, group: group)
         let promise: EventLoopPromise<IOError> = serverChannel.eventLoop.newPromise()
 
-        XCTAssertNoThrow(try serverChannel.pipeline.add(handler: AcceptHandler(promise)).then {
-            serverChannel.register()
-        }.then {
-            serverChannel.bind(to: try! SocketAddress(ipAddress: "127.0.0.1", port: 0))
-        }.wait())
+        XCTAssertNoThrow(try serverChannel.eventLoop.submit {
+            serverChannel.pipeline.add(handler: AcceptHandler(promise)).then {
+                serverChannel.register()
+            }.then {
+                serverChannel.bind(to: try! SocketAddress(ipAddress: "127.0.0.1", port: 0))
+            }
+        }.wait().wait() as Void)
 
         XCTAssertEqual(active, try serverChannel.eventLoop.submit {
             serverChannel.readable()

